src/Mobs/Monster.cpp

```diff
@@ -292,48 +269,48 @@ void cMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		Vector3d Distance = m_Destination - GetPosition();
 		if (!ReachedDestination() && !ReachedFinalDestination())  // If we haven't reached any sort of destination, move
 		{
-			Distance.y = 0;
-			Distance.Normalize();
-
-			if (m_bOnGround)
-			{
-				Distance *= 2.5f;
-			}
-			else if (IsSwimming())
-			{
-				Distance *= 1.3f;
-			}
-			else
+			if (m_Destination != GetPosition())
```

Comment:
It is possible that this is always returning true, this might not be needed at all, I'll check it out.